src/cargo/ops/cargo_new.rs

```diff
@@ -17,11 +17,13 @@ use toml;
 #[derive(Clone, Copy, Debug, PartialEq)]
 pub enum VersionControl { Git, Hg, NoVcs }
 
+#[derive(Clone, Debug, PartialEq)]
 pub struct NewOptions<'a> {
     pub version_control: Option<VersionControl>,
     pub bin: bool,
     pub path: &'a str,
     pub name: Option<&'a str>,
+    pub sourcefile_relative_path : Option<String>,
```

Comment:
In this case `mk` should accept not `NewOptions`, but something else. Shall I create special private struct like `MkOptions` and convert from `NewOptions` to `MkOptions` both in `new` and in `init`?